table/tables/tables.go

```diff
@@ -129,7 +129,28 @@ func (t *Table) Meta() *model.TableInfo {
 
 // Cols implements table.Table Cols interface.
 func (t *Table) Cols() []*column.Col {
```

Comment:
you can cache this and following writeable cols for later use. 

Btw, does outer need to know write only? I suggest handling all in table.